src/Illuminate/Events/Dispatcher.php

```diff
@@ -72,7 +72,7 @@ public function __construct(ContainerContract $container = null)
      *
      * @param  \Closure|string|array  $events
      * @param  \Closure|string|array|null  $listener
-     * @return void
+     * @return void|\Illuminate\Support\Collection
```

Comment:
Void can never be combined with something else. This method follows its contract which has void. The only reason why there are returns is to early terminate the method call.

Reply:
thank you for your advice.
I think if we want to be on contract its better to use this :
```
 if ($events instanceof Closure) {
            collect($this->firstClosureParameterTypes($events))
                ->each(
                    fn($event) => $this->listen($event, $events)
                );

            return;
        } elseif ($events instanceof QueuedClosure) {
```

I reopen this again and changed the PR

Reply:
Changing it has no tangible benefit.

Reply:
IDE shows this code doc block as a warning, and the behavior of this method is different from its return type and its contact.